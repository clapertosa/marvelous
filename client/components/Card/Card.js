import styles from "./Card.scss";

const Card = props => {
  return (
    <div className={styles["grid-container"]}>
      <div style={{ maxWidth: "150px" }} className={styles["image-container"]}>
        <img
          style={{ width: "100%" }}
          src="https://terrigen-cdn-dev.marvel.com/content/prod/1x/_001avn_com_crd_01.jpg"
          alt=""
        />
      </div>
      <div className={styles.caption}>
        <div className={styles.main}>Doctor Strange</div>
        <div className={styles["sub-main"]}>Stephen Strange</div>
      </div>
    </div>
  );
};

export default Card;
