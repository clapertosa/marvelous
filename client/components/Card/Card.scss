.grid-container {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: auto 1fr;
  grid-template-areas: "image" "caption";
  transition: transform 0.3s ease-out;

  &:hover {
    .caption {
      color: red;
      transition: color 0.8s;
    }
    overflow: hidden;
    transform: scale(1.2);
  }

  a {
    color: white;
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    text-decoration: none;
  }
}

.image-container {
  grid-area: image;
  max-width: 150px;

  img {
    width: 100%;
    height: auto;
  }
}

.caption {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: 1fr 1fr;
  grid-template-areas: "main main main" "sub-main sub-main sub-main";
  grid-area: caption;
  text-align: center;
  font-weight: bold;
  font-family: "Marvel-bold";
  width: 100%;
}

.main {
  grid-area: main;
}

.sub-main {
  grid-area: sub-main;
}
