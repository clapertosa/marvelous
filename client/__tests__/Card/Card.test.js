import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import Card from "../../components/Card/Card";

describe("Card", () => {
  it("renders", () => {
    expect(
      shallow(
        <Card
          image="/img/image"
          imageExtension="jpg"
          title="Beauty and the Beast"
        />
      )
    );
  });

  it("matches the snapshot", () => {
    const wrapper = shallow(
      <Card
        image="/img/image"
        imageExtension="jpg"
        title="Beauty and the Beast"
      />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.setProps({ title: "Beauty and the Best" });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
