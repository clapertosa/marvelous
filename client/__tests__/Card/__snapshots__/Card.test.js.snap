// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapshot 1`] = `
<div
  className=" "
>
  <Link
    as="/undefined/undefined"
    href="/undefined?id=undefined"
    prefetch={true}
  >
    <a>
      <div>
        <img
          alt="Beauty and the Beast thumbnail"
          src="/img/image/portrait_uncanny.jpg"
        />
      </div>
      <div>
        <div>
          Beauty and the Beast
        </div>
      </div>
    </a>
  </Link>
</div>
`;

exports[`Card matches the snapshot 2`] = `
<div
  className=" "
>
  <Link
    as="/undefined/undefined"
    href="/undefined?id=undefined"
    prefetch={true}
  >
    <a>
      <div>
        <img
          alt="Beauty and the Best thumbnail"
          src="/img/image/portrait_uncanny.jpg"
        />
      </div>
      <div>
        <div>
          Beauty and the Best
        </div>
      </div>
    </a>
  </Link>
</div>
`;
